### PR TITLE
Support only MULTI_ROLE frameworks

### PR DESCRIPTION
**Pull request by alenkacz**
MULTI_ROLE check is in MesosClient as well as USI core because one is not tight to our MesosClient implementation and there can be alternative ones. At the same time, USI right now won't work on properly for single role framework because for those `AllocationInfo` is optional.

JIRA: DCOS-47475

### CODE REVIEW

File: core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala [hunk 1]
```diff
@@ -50,9 +52,16 @@ object Scheduler {
   type StateOutput = (StateSnapshot, Source[StateEvent, Any])
 
   def fromClient(client: MesosClient): Flow[SpecInput, StateOutput, NotUsed] = {
+    if (!isMultiRoleFramework(client.frameworkInfo)) {
+      throw new IllegalArgumentException(
+        "Mesos client supports only MULTI_ROLE frameworks. Please provide create MesosClient with FrameworkInfo that has capability MULTI_ROLE")
```
Comment (zen-dog) [inline core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala]: It's rather: `USI scheduler provides support for MULTI_ROLE frameworks only....`

File: mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala [hunk 2]
```diff
@@ -294,6 +300,11 @@ object MesosClient extends StrictLogging with StrictLoggingFlow {
       system: ActorSystem,
       materializer: ActorMaterializer): Source[MesosClient, NotUsed] = {
 
+    if (!isMultiRoleFramework(frameworkInfo)) {
```
Comment (zen-dog) [inline mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala]: I would not put this check into the client. It's the "limitation" of our core-scheduler.

File: core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala [hunk 3]
```diff
@@ -50,9 +52,16 @@ object Scheduler {
...
+    }
     fromFlow(client.calls, Flow.fromSinkAndSource(client.mesosSink, client.mesosSource))
   }
 
+  private def isMultiRoleFramework(frameworkInfo: FrameworkInfo): Boolean =
+    frameworkInfo.getCapabilitiesList.asScala.exists(c => c.getType == FrameworkInfo.Capability.Type.MULTI_ROLE)
```
Comment (jeschkies) [inline core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala]:
```suggestion
    frameworkInfo.getCapabilitiesList.asScala.exists(_.getType == FrameworkInfo.Capability.Type.MULTI_ROLE)
```

File: mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala [hunk 4]
```diff
@@ -25,6 +26,11 @@ trait MesosClient {
     */
   def frameworkId: FrameworkID
 
+  /**
+    * The frameworkInfo as which this client is currently subscribed with.
```
Comment (jeschkies) [inline mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala]:
```suggestion
    * The framework info this client is currently subscribed with.
```

**Review by jeschkies — COMMENTED** [review]
Do we have a JIRA?

File: mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala (hunk 2 above)
Comment (alenkacz) [reply mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala]: okay, makes sense!

**Review by zen-dog — APPROVED** [review]

**Review by jeschkies — APPROVED** [review]

**Pull request merged**